libs/vscode/nx-project-view/src/lib/views/nx-project-base-view.ts

```diff
@@ -9,34 +8,23 @@ import {
 } from '../nx-project-tree-item';
 
 export type ProjectViewStrategy<T> = Required<
-  Pick<TreeDataProvider<T>, 'getChildren' | 'getParent'>
+  Pick<TreeDataProvider<T>, 'getChildren'>
 >;
 
-export abstract class BaseView {
-  constructor(protected readonly cliTaskProvider: CliTaskProvider) {}
-
-  async getParentOfTargetItem(element: NxTargetTreeItem) {
-    const { nxProject, nxTarget } = element;
-    const projectDef = (await this.cliTaskProvider.getProjects())[
-      nxProject.project
-    ];
-
-    if (!nxTarget.configuration) {
-      return this.createProjectTreeItem([nxProject.project, projectDef]);
+type RootPath = string;
+export type ProjectDefinition =
+  | {
+      [projectName: string]: ProjectConfiguration;
     }
+  | { [projectName: string]: RootPath };
 
-    if (!projectDef.targets || !projectDef.targets[nxTarget.name]) {
-      getOutputChannel().appendLine(
-        `Could not find target '${nxTarget.name}' in project '${nxProject.project}', even though it should exist`
-      );
-      return null;
-    }
+export interface ProjectInfoProvider {
+  getWorkspacePath(): string;
+  getProjects(): Promise<ProjectDefinition>;
+}
 
-    return this.createTargetTreeItem(nxProject, [
-      nxTarget.name,
-      projectDef.targets[nxTarget.name],
-    ]);
-  }
+export abstract class BaseView {
+  constructor(protected readonly infoProvider: ProjectInfoProvider) {}
```

Comment:
Can you tell me about your motivation behind the `infoProvider` abstraction? Since we still have to inject `CliTaskProvider` in the different strategies, I don't really see the upside...

Reply:
The motive was to create an interface that already has the correct typings. But, as this will be addressed separately, I reverted the change.